integration/clean/clean_aws_resource.go

```diff
@@ -0,0 +1,145 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build clean
+// +build clean
+
+package main
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Type "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	ssmType "github.com/aws/aws-sdk-go-v2/service/ssm/types"
+	smithyTime "github.com/aws/smithy-go/time"
+)
+
+const (
+	daysToKeep   = 30
+	keepDuration = -1 * time.Hour * 24 * time.Duration(daysToKeep)
+)
+
+var expirationDate = time.Now().UTC().Add(keepDuration)
```

Comment:
Rather than use a global here, can you instantiate the expiration date inside of the `main` function, and then pass it in as an argument to the functions it needs to go to?

Reply:
The reason I didn't ask for this in the original PR was because it looked like we were only going to be cleaning AMIs. Since you're extending the functionality of the script, I figured now would be a good time to parameterize rather than having a global value that gets instantiated when the main gets invoked.

Reply:
True though. I intend to separate the go files and parameterize if we intend to extends the function more, same as `aws-otel-collector` does. However, I'm good with doing now